### ISSUE TITLE
Release Android 2.2.0

### ISSUE THREAD

**Issue by bleege** [mapbox/mapbox-gl-native#2847]
It's been just [a week since the original 2.1.0 release](https://github.com/mapbox/mapbox-gl-native/issues/2689), but we have all sorts of good stuff that we should get out into the world.

### New Highlights

* [New User Location Dot](https://github.com/mapbox/mapbox-gl-native/issues/2732)
* [Support For Custom Images For Markers](https://github.com/mapbox/mapbox-gl-native/issues/2744)
* [Default Marker](https://github.com/mapbox/mapbox-gl-native/issues/2506)
* [OpenGL fixes](https://github.com/mapbox/mapbox-gl-native/issues/2817) to support Ice Cream Sandwich (API 15) and Jelly Bean (API 16, 17, 18) devices
* [Follow User Location Dot](https://github.com/mapbox/mapbox-gl-native/issues/2758)
* [Bulk Add Markers](https://github.com/mapbox/mapbox-gl-native/issues/2726)
* [Satellite Streets Style](https://github.com/mapbox/mapbox-gl-native/issues/2739)

Full Change Log: https://github.com/mapbox/mapbox-gl-native/issues?q=is%3Aissue+milestone%3Aandroid-v2.2.0

Release Branch: https://github.com/mapbox/mapbox-gl-native/commits/release-android-v2.2.0

### Here's the TODO list for making that happen:

- [x] Change Version Name to `2.2.0`
- [x] Publish `2.2.0` artifact to Maven Central
- [x] Updating Mapbox.com JavaDoc Documentation
- [x] Tag `android-v2.2.0` in GitHub
- [x] Update version info in Twitter Fabric metadata file
- [x] Move open issues to `android-v2.3.0` Milestone in GitHub
- [ ] Close Milestone `android-v2.2.0` in GitHub

/cc @mapbox/mobile @mapbox/support @twbell

**Comment by bleege** [comment]
FYI @ljbade @tobrun I've moved the milestone of tickets that were associated with commits that landed in `master` since the release branch commit 43dde0f204ac770feae0ef22c7a4ea67585deb12 on 27-Oct-2015.  The commits are still in `master`, but the milestone was changed from `android-v2.2.0` to `android-v2.3.0` as they won't make this release.  Below is a screenshot of the affected commits intermingled with other non Android commits:

![screen shot 2015-10-28 at 4 46 58 pm](https://cloud.githubusercontent.com/assets/989928/10804125/e0f4b87e-7d93-11e5-8ab7-f7d7c51a635d.png)

**Comment by ljbade** [comment]
@bleege Since the switch over from `Marker.setSprite(String)` to `Marker.setIcon(Sprite)` is breaking, should we go to 3.0.0?

**Comment by bleege** [comment]
@ljbade Can you clarify by what you mean by "breaking"?  Do you mean that `setSprint()` was renamed to `setIcon()`?

**Comment by ljbade** [comment]
@bleege It is not just a simple rename, we also have a new way of loading sprites. Instead of a `string` which points to the style specific sprite sheet (and was really just a temporary solution - iOS does not support this for a good reason), you now first load a `Sprite` via `SpriteFactory`.

**Comment by ljbade** [comment]
So code using the old `setSprite` will need to be changed, and the app developer will have to add a Bitmap resource with the icon they want. Whether it be from Maki, or their own design.

**Comment by bleege** [comment]
Thanks for clarifying @ljbade.  Here's what I'm thinking:

> It is not just a simple rename, we also have a new way of loading sprites.

This is to be understood and fine as this is the first release were the SDK will offer custom sprite loading.  It's also the second overall release of the SDK (with the first one being just a week ago).  Looking at the code the impact will be limited to `MarkerOptions.icon()` as [`Marker.setIcon()`](https://github.com/mapbox/mapbox-gl-native/blob/9361669fb3cc682a807e8b8618f0586839a066ba/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Marker.java#L64-L69) is not directly accessible to developers (and has a friendly JavaDoc note to not directly use this  :smiley: ).  The end result is that to "upgrade" will require the following change (assuming a custom Style Sprite was even specified and not just using the default):

```java
// 2.1.0
new MarkerOptions().sprite()

// 2.2.0
new MarkerOptions().icon()
```

In short, this is still the very early days of the SDK and API changes are to be expected.  This one isn't large enough to warrant going up a full [semver](http://semver.org/).

**Comment by bleege** [comment]
Just finished publishing 2.2.0 to Maven Central.  It'll take a few hours for it to show up in http://search.maven.org/ though.

**Comment by bleege** [comment]
Tagged as `android-v2.2.0` and pushed.  Here's the commit tree for this tag:

https://github.com/mapbox/mapbox-gl-native/commits/android-v2.2.0

![screen shot 2015-10-29 at 10 42 16 am](https://cloud.githubusercontent.com/assets/989928/10823444/c075bfc6-7e29-11e5-99d9-5cc8f5af89a2.png)

**Comment by bleege** [comment]
Finally landed in Maven Central -->  https://repo1.maven.org/maven2/com/mapbox/mapboxsdk/mapbox-android-sdk/2.2.0/

**Comment by bleege** [comment]
Confirmed that 2.2.0 from Maven Central works as expected by loading into local copy of Android Demo App.

![device-2015-10-29-110112](https://cloud.githubusercontent.com/assets/989928/10824050/a8ad8128-7e2c-11e5-805d-4fb150315404.png)

**Comment by bleege** [comment]
Just published the JavaDoc updates to https://www.mapbox.com/android-sdk/.  Waiting for it to regenerate which should just take a few minutes.

**Comment by bleege** [comment]
JavaDocs are live!

Time to close this milestone and call it!

:beers:

**Comment by Mansi26** [comment]
Hi..i am trying to center the Info window when the map is zoomed.
But ..i could not find any listener for zoom event.
Can anyone help..regarding how to do this...?
